spec/game_config_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'require_all'
+require_all 'lib'
+
+describe GameConfig do
+
+  let(:game_config) { GameConfig.new }
+  
+  describe "#set_up" do
+
+    it "returns a hash with a key-value pair for board:, rules:, user_interface:, and players:" do 
+      allow(game_config).to receive(:player_config).and_return(["Player 1", "Player 2"])
```

Comment:
It's a good rule to never stub the thing you're testing. It looks like `player_config` should be passed in somehow.

Reply:
Updated GameConfig, PlayerConfig, and tests to avoid this. Also renamed PlayerConfig to PlayerFactory to make responsibility clearer.